tests/test_utils.py

```diff
@@ -184,3 +184,18 @@ def test_sanitize_list_handling(self):
         encoder = SanitizingJSONEncoder()
         # no assert as we are just expecting this not to throw
         encoder._sanitize(data, ['password'], ['password'])
+
+    def test_json_encode_invalid_keys(self):
+        """
+        Test that _sanitize can accept some invalid json where a function
+        name or some other bad data is passed as a key in the payload dictionary.
+        """
+        def foo():
+            return "123"
```

Comment:
We'll need to expand the number of test cases for other potentially invalid keys.

Reply:
Yea, I didn't have any immediate good ideas on other test cases where a key would be invalid but wouldn't be handled by the json encoder. For example, ints and None are handled internally and dont cause exceptions.